packages/zod/src/zod.test.ts

```diff
@@ -668,6 +668,201 @@ describe('generatePartOfSchemaGenerateZod', () => {
   });
 });
 
+const apiSchemaWithDiscriminator: GeneratorOptions = {
+  pathRoute: '/dogs',
+  context: {
+    specKey: 'dog',
+    specs: {
+      dog: {
+        openapi: '3.0.0',
+        info: {
+          version: '1.0.0',
+          title: 'Dogs',
+        },
+        paths: {
+          '/dogs': {
+            get: {
+              summary: 'List all dogs',
+              operationId: 'listDogs',
+              tags: ['dogs'],
+              responses: {
+                '200': {
+                  description: 'A paged array of dogs',
+                  content: {
+                    'application/json': {
+                      schema: {
+                        $ref: '#/components/schemas/Dogs',
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+        components: {
+          schemas: {
+            Dogs: {
+              type: 'array',
+              items: {
+                $ref: '#/components/schemas/Dog',
+              },
+            },
+            Dog: {
+              type: 'object',
+              oneOf: [
+                {
+                  $ref: '#/components/schemas/Labradoodle',
+                },
+                {
+                  $ref: '#/components/schemas/Dachshund',
+                },
+              ],
+              required: ['type'],
+              properties: {
+                barksPerMinute: {
+                  type: 'integer',
+                },
+              },
+              discriminator: {
+                propertyName: 'breed',
+                mapping: {
+                  Labradoodle: '#/components/schemas/Labradoodle',
+                  Dachshund: '#/components/schemas/Dachshund',
+                },
+              },
+            },
+            Labradoodle: {
+              type: 'object',
+              required: ['cuteness'],
+              properties: {
+                cuteness: {
+                  type: 'integer',
+                },
+                // in the real runner breed is added by getApiSchemas in import-open-api.ts, inferred from the discriminator
```

Comment:
its unfortunate that i can't do a full integration style test with this because there is some pre-processing of the schema in the core code which handles the expanding of the breed properties onto the types specifies in the `discriminator`